### DISCUSSION TITLE
A WORKING SOCIAL MEDIA SITE I BUILT! (1.5+ years of development)

### DISCUSSION THREAD

**Discussion by eesazahed** [supabase/supabase#21241] [Show and tell]
Hi everyone! 

I started developing a social media site as a hobby project back in the summer of 2022 (when I was in 8th grade). Here I am in 10th grade with a somewhat completed and functional social media site.

https://laser.vercel.app

Let me know what you think! If you find any bugs, please let me know, and I'll try to figure something out. Right now I just want people to test the app out.

FAQ's:
- **What language/framework**: Next.js + typescript
- **Database**: Supabase (free postgres)
- **Hosting**: Vercel
- **Auth**: Google
- **Why is it slow**: The initial load takes a few seconds because of server-side-rendering. It's faster after that.
- **Is there a profanity filter?** No because I realized people would always find a way to get past it somehow
- **Why can't I post images?** For various reasons. Just allowing text for now.
- **Can I get banned?** Yes, your account can be temporarily or permanently disabled.
- **Can I view the code?** Not at the moment.
- **Is it open source?** No, for now.
- **I can't edit my posts?** Yes, to preserve authenticity
- **How does caching work?** I chose not to really use caching, instead server-side rendering and fetch small amounts of information at a time for the most accurate and up-to-date results.
- **How long did it take to make?** 1.5 years, however I'd probably work 2-3 hours a week, and sometimes not at all. If I really grinded I could have made it in probably a month if I worked 8 hours a day... maybe.
- **Are you paying for this?** No I'm a broke highschooler lol. If this site starts actually getting enough users so that I would need to pay money to maintain it, I would probably have to figure something out. Idk.

Alright everyone, enjoy and give me feedback!
​​
​​

Edit Post

**Comment by Meltingoreos** [comment]
Hey, another 10th grader in high school here

I decided to get into CS pretty recently, and I have near no coding experience except for the fact that I am taking AP CSP (Computer Science Principles) That teaches using javascript(via code.org or codehs). I'm in a pretty big rush to create an online portfolio for colleges and stuff. I really like the coding aspect of the class, and am thinking of being a software developer or maybe a game dev. Do you have any advice or something like that? Thanks in advance

**Reply by eesazahed** [reply]
Hey whats up? Congrats on getting started on your coding journey!

So... are you looking for advice to create a portfolio or just for software development in general?
